datasets/lt/pep_declarations/crawler.py

```diff
@@ -0,0 +1,199 @@
+            else:
+                self.context.log.error(
+                    f"deklaracija {id_str} skipped due to {status_code} error."
+                )
+
+
+def make_person(context: Context, data: dict) -> Entity:
+    person = context.make("Person")
+    first_name = data.pop("vardas")
+    last_name = data.pop("pavarde")
+    person_id = data.pop("asmensKodas", None)  # this identifier is often missing
+    person.id = context.make_id(person_id, first_name, last_name)
+    person.add("firstName", first_name)
+    person.add("registrationNumber", person_id)
+    person.add("lastName", last_name)
+    person.add("birthDate", data.pop("gimimoData", None))
+    person.add("legalForm", data.pop("asmensTipas", None))
+    context.audit_data(data)
+    return person
+
+
+def make_marriage(context: Context, person: Entity, spouse: Entity) -> Entity:
+    marriage = context.make("Family")
+    marriage.id = context.make_id(person.id, spouse.id)
+    marriage.add("relationship", "Spouse")
+    marriage.add("person", person)
+    marriage.add("relative", spouse)
+    return marriage
+
+
+def parse_affiliations(
+    context: Context, person: Entity, affiliations: list[dict], is_pep: bool = True
+) -> list[tuple[Entity]]:
+    """
+    Args:
+        context (Context)
+        affiliations (list[dict]): A list of dicts, where each dict refers to a place
+         of work and a sub-list of positions.
+        is_pep (bool, optional): Defaults to True.
+
+    Returns:
+        list[tuple[Entity]]: a flattened list of tuples, where each is
+        a position and occupancy. Occupancies that do not meet OpenSanctions criteria
+        are skipped.
+    """
+    entities = []
+    for data in affiliations:
+        entity_is_legal_entity: bool = data.pop("yraJuridinisAsmuo")
+        entity_is_lithuanian: bool = data.pop("registruotaLietuvoje")
+        legal_entity_code: Optional[str] = data.pop("jaKodas")
+        entity_name = data.pop("pavadinimas")
+        affiliation_start_date: str = data.pop("rysioPradzia")
+        affiliation_type = data.pop("darbovietesTipas")
+        data_sources = data.pop("duomenuSaltiniai")
+        filed_automatically: bool = data.pop("uzpildytaAutomatiskai")
+        entity_must_be_declared: bool = data.pop("privaluDeklaruoti")
+        entity_legal_form: str = data.pop("jaTeisinesFormosPavadinimas")
+
+        for role in data.pop("pareigos"):
+            position_name: Optional[str] = role.pop("pareigos")
+            legal_code: Optional[str] = role.pop("teisejoKodas")
+            role_must_be_declared: Optional[bool] = role.pop("privaluDeklaruoti")
+            nature_of_duties: str = role.pop("pareiguTipasPavadinimas")
+            position = h.make_position(
+                context,
+                name=position_name or "Unknown position",
+                topics=None,
+                country="LT" if entity_is_lithuanian else None,
```

Comment:
The JSON has an `is_lithuanian` key but the portal displays "country of registration" which would lead you to believe that they actually give you the country if it's foreign, but they don't. I've only seen  "Republic of Lithuania" or "foreign country".

I've seen some where `is_lithuanian` is False ([example](https://pinreg.vtek.lt/app/pid-perziura/608829) -- under 'Registracijos šalis: Užsienio valstybė'). 
Usually for the spouse, sometimes for the declarant. 
[An example from a spouse.](https://pinreg.vtek.lt/app/pid-perziura/609590)
It would be bizzare if the 'main' data was from a foreign entity. I imagine in the first case I linked it was not a requirement to declare. I haven't seen an example where a non-lithuanian entity could be significant in the context of OS criteria.

Reply:
ok, this diff was quite interesting. Could you add it without the print?:

```diff
         for role in data.pop("pareigos"):
-            position_name: str = role.pop("pareigos") or "Unknown position"
+            position_name: Optional[str] = role.pop("pareigos")
             legal_code: Optional[str] = role.pop("teisejoKodas")
             role_must_be_declared: Optional[bool] = role.pop("privaluDeklaruoti")
+            print(
+                f"e={entity_must_be_declared} r={role_must_be_declared} l={entity_is_lithuanian}",
+                "%s, %s" % (position_name, entity_name),
+            )
+            assert (not role_must_be_declared) or position_name, (
+                person,
+                role_must_be_declared,
+                position_name,
+            )
+            assert (not role_must_be_declared) or entity_is_lithuanian, (
+                person,
+                role_must_be_declared,
+                entity_is_lithuanian,
+            )
+
             nature_of_duties: str = role.pop("pareiguTipasPavadinimas")
+            if not role_must_be_declared:
+                continue
             position = h.make_position(
```

for https://pinreg.vtek.lt/app/pid-perziura/608829 this gives me

```
e=True r=True l=True Direktorius, Kauno technologijos universitetas
e=True r=False l=True Profesorius, Kauno technologijos universitetas
e=True r=False l=True Vyriausiasis mokslo darbuotojas, Kauno technologijos universitetas
e=False r=False l=True Vyr. tyrėjas, MB Open technology
e=False r=False l=True Profesorius, Vytauto Didžiojo universitetas
e=False r=False l=False Adjunktas, Politechnika Śląska
```

i.e. only the director post was sufficiently senior to need to be declared.

I haven't seen a role_must_be_declared that wasn't lithuanian.

Then we can set the country as lithuanian with confidence.

It also means we skip adding lots of low-ranking positions - all we care about are the positions where they control public money, which happens to be the ones that require declaration in this register.

Some other interesting examples:

```
e=False r=None lTrue vyriausiasis tyrėjas, Valstybės sienos apsaugos tarnyba prie Lietuvos Respublikos vidaus reikalų ministerijos
e=False r=None lTrue Chief Investigator, State Border Guard Service under the Ministry of Internal Affairs of the Republic of Lithuania

e=True r=True lTrue Vyriausioji specialistė, Lietuvos kalėjimų tarnyba
e=True r=True lTrue Senior Specialist, Lithuanian Prison Service

e=False r=None None None, Viešoji įstaiga Plungės rajono savivaldybės ligoninė
e=False r=None None None, Public institution Plungė district municipal hospital

e=True r=True lTrue Vyresnysis inžinierius technologas, AB "LTG Cargo"
e=True r=True lTrue Senior Technologist, AB "LTG Cargo"
Shareholders 100% of shares are owned by AB Lietuvos Geležinkeliai
 https://doc.ltg.lt/en/About%20us/Reports/2020/Annual/AB%20LTG%20Cargo%20annual%20report%202020.pdf
Lithuanian Railways, is the national state-owned railway company of Lithuania.
```